sources/osg/KdTree.js

```diff
@@ -378,15 +427,72 @@ KdTree.prototype = MACROUTILS.objectLibraryClass( {
     setVertices: function ( vertices ) {
         this._vertices = vertices;
     },
-    getNodes: function () {
-        return this._kdNodes;
-    },
     getTriangles: function () {
         return this._triangles;
     },
     setTriangles: function ( triangles ) {
         this._triangles = triangles;
     },
+    setPrimitiveIndices: function ( indices ) {
+        this._primitiveIndices = indices;
+    },
+    getPrimitiveIndices: function () {
+        return this._primitiveIndices;
+    },
+
+    // vector containing the primitive vertex index data packed as no_vertice_indices then vertex indices 
+    // ie. for points it's (1, p0), for lines (2, p0, p1) etc.
+    setVertexIndices: function ( indices ) {
+        this._vertexIndices = indices;
+    },
+
+    getVertexIndices: function () {
+        return this._vertexIndices;
+    },
+
+
+    getNodes: function () {
+        return this._kdNodes;
+    },
+
+    addPoint: function ( p0 ) {
+        var i = this._vertexIndices.length;
+        this._primitiveIndices.push( i );
+        this._vertexIndices.push( 1 );
+        this._vertexIndices.push( p0 );
+        return i;
+    },
+
+    addLine: function ( p0, p1 ) {
+        var i = this._vertexIndices.length;
+        this._primitiveIndices.push( i );
```

Comment:
Ideally _primitivesIndices and _vertexIndices size should be computed first so that we can use a typed array and avoid the multiple allocation (push).

Depending of the size, _primitiveIndices or _vertexIndices could use Uint16Array too to reduce memory footprint of the kdtree.